cloud/cluster/memory/cluster.go

```diff
@@ -0,0 +1,108 @@
+package memory
+
+import (
+	"github.com/scootdev/scoot/cloud/cluster"
+)
+
+func NewCluster(initial []cluster.Node, updateCh chan []cluster.NodeUpdate, wrapper cluster.NodeWrapper) cluster.Cluster {
+	if wrapper != nil {
+		for i, member := range initial {
+			initial[i] = wrapper.Wrap(member)
+		}
+	}
+
+	c := &simpleCluster{
+		inCh:    updateCh,
+		reqCh:   make(chan interface{}),
+		wrapper: wrapper,
+		members: State{members: initial},
+		subs:    nil,
+	}
+	go c.loop()
+	return c
+}
+
+type simpleCluster struct {
+	inCh  chan []cluster.NodeUpdate
+	reqCh chan interface{}
+
+	wrapper cluster.NodeWrapper
+	members State
+	subs    []chan []cluster.NodeUpdate
+}
+
+func (c *simpleCluster) Members() ([]cluster.Node, error) {
+	ch := make(chan []cluster.Node)
+	c.reqCh <- ch
+	return <-ch, nil
+}
+
+func (c *simpleCluster) Subscribe() (cluster.Subscription, error) {
+	ch := make(chan cluster.Subscription)
+	c.reqCh <- ch
+	return <-ch, nil
+}
+
+func (c *simpleCluster) Close() error {
+	close(c.reqCh)
+	return nil
+}
+
+func (c *simpleCluster) done() bool {
+	return c.inCh == nil && c.reqCh == nil
+}
+
+func (c *simpleCluster) loop() {
+	for !c.done() {
+		select {
+		case updates, ok := <-c.inCh:
+			for i, _ := range updates {
+				if updates[i].Node != nil {
+					updates[i].Node = c.wrapper.Wrap(updates[i].Node)
+				}
+			}
+			if !ok {
+				c.inCh = nil
+				continue
+			}
+			c.members.Apply(updates)
+			for _, sub := range c.subs {
+				sub <- updates
```

Comment:
I feel like we should create a goroutine for each of these. It might be ok to block if there's only one subscriber allowed, but with multiple it seems iffy.